sdk/data/azcosmos/emulator_cosmos_global_endpoint_manager_test.go

```diff
@@ -73,3 +73,68 @@ func TestGlobalEndpointManagerEmulator(t *testing.T) {
 	assert.Equal(t, len(locationInfo.availReadEndpointsByLocation), len(availableEndpointsByLocation)+1)
 	assert.Equal(t, len(locationInfo.availWriteEndpointsByLocation), len(availableEndpointsByLocation)+1)
 }
+
+func TestGlobalEndpointManagerPolicyEmulator(t *testing.T) {
```

Comment:
How is this tests testing the policy? Based on the code, it is just accessing the GEM within the client and manually execting GetAccountProperties (which is not a bad test) but it does not verify that if a user does an operation (for example, ReadItem) the policy is calling GetAccountProperties in the back?

Maybe if you replace the GetAccountProperties line for a normal operation on the client, the other asserts can remain in place?

Reply:
Or inject a policy on client creation (like in some of the unit tests) to verify that there was an HTTP request made for the account policies?

Reply:
Alternatively, adding unit tests for the policy itself could also work (like we have for https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/cosmos_headers_policy_test.go)

Reply:
I am running an http request to populate the account information - the information is only populated after the create database call that is being made. If you'd like for me to remove some of the checks being done before the request is made I can, but what you mentioned in your first point here is exactly what I am testing.
<img width="550" alt="image" src="https://github.com/Azure/azure-sdk-for-go/assets/30335873/d1fb0e8c-f7f1-42dd-b43a-b4693f9d180b">

Reply:
My bad, I got confused. It seems the piece of code with the gem.GetAccountProperties is really not part of the testing of the policy